[Elixir] New concept exerise: `enum` (#1587)
exercism/v3: languages/elixir/exercises/concept/enum/.docs/hints.md

## General

- Read about the `Enum` module in the [official Getting Started guide][getting-started-enum] or on [elixirschool.com][elixir-school-enum]
- Take a look in the documentation for the full [list of functions in the `Enum` module][enum-fuctions].

## 1. Sort items by price

- There is a [built-in function][enum-sort-by] for sorting enumerables using a sorter function.

## 2. Find all items with missing prices

- There is a [built-in function][enum-filter] for filtering enumerables.

## 3. Increment the item's quantity

> - Maps are enumerables too.
> - Iterating over a map will convert the map to a list of `{key, value}` tuples.

Reviewer: ```suggestion
- Maps implement the enumerable protocol.
- `Enum` functions convert maps to a list of `{key, value}` tuples.
```

- There is a [built-in function][enum-map] for replacing every element in an enumerable with another element.
- There is a [built-in function][enum-into] that can transform a list of `{key, value}` tuples back into a map.

## 4. Calculate the item's total quantity

- Maps are enumerables too.

> - Iterating over a map will convert the map to a list of `{key, value}` tuples.

Reviewer: Caution repeating the same hint.

@ErikSchierboom, @iHiD is the thought with hints that only one hint will be displayed at once? or will you see them all at the same time?

Reviewer 2: Excellent question! I don't think we've decided upon this. I think students will be able to show hints per tasks.

Author: I'll keep the repeating hints then.

It would be really good to only show hints per task so that the whole exercise doesn't get spoiled if the student only has trouble with starting it.

Reviewer 3: > I think students will be able to show hints per tasks.

Yep - that's the idea.
